std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoHTTPResponseCache.java

```diff
@@ -0,0 +1,182 @@
+package org.enso.base.enso_cloud;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.http.HttpHeaders;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Optional;
+import org.enso.base.cache.LRUCache;
+import org.enso.base.cache.ResponseTooLargeException;
+
+/**
+ * EnsoHTTPResponseCache is a cache for EnsoHttpResponse values that respects the cache control HTTP
+ * headers received in the original repsonse to a request.
+ *
+ * <p>It uses LRUCache, so it also puts limits on the size of files that can be requested, and on
+ * the total cache size, deleting entries to make space for new ones. All cache files are set to be
+ * deleted automatically on JVM exit.
+ *
+ * <p>Without caching, EnsoHttpResponse contains an InputStream providing the response data. When
+ * there is a cache hit, this stream reads from the local file storing the cached data. When there
+ * is no cache hit, the InputStream is connected directly to the remote server.
+ */
+public class EnsoHTTPResponseCache {
+  private static final int DEFAULT_TTL_SECONDS = 31536000;
```

Comment:
During the refactor it seems that the comment specifying how long time interval that number represents got lost.

I think that comment was very helpful.